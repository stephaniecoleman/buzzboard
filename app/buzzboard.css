.board {
  margin: 20px 50px;
  text-align: center;
}

.boardTile {
  margin: 10px;
}
