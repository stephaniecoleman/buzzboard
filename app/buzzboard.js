import React from 'react';

// MODEL

export const initialState = {
  isPlaying: false,
  currentSound: null
};

// UPDATE

export const toggleSound = (newSound) => (
  { type: 'TOGGLE', newSound }
);

export const reducer = (state, action) => {
  switch (action.type) {
  case 'TOGGLE':
    const isPlaying = !state.isPlaying || state.currentSound !== action.newSound
    return {...state, isPlaying: isPlaying, currentSound: action.newSound}
  default:
    return state;
  }
};

// VIEW

const Tile = ({sound, onTileClick}) => {
  return (
    <div onClick={onTileClick(sound)}>
      <h3>{sound.title}</h3>
    </div>
  );
};

export const Board = ({sounds, onTileClick}) => {
  return (
    <div id='board'>
      {sounds.map((sound) => {
        return <Tile
          key={sound.file}
          sound={sound}
          onTileClick={onTileClick}
        />
      })}
    </div>
  );
};

// WHERE DOES THIS STUFF GO?

const player = document.getElementById('player');

const playSound = (sound) => () => {
  player.setAttribute('src', sound);
  player.play();
};
