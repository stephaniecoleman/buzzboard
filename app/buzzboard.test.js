import renderer from 'react-test-renderer';
import React from 'react';
import { toggleSound, initialState } from './buzzboard.js'

describe('toggleSound', () => {
  const action = { 'newSound': 'myCoolNewSound', 'type': 'TOGGLE' }
  it('returns an action', () => {
    expect(toggleSound('myCoolNewSound')).toEqual(action);
  })
})

// MODEL

describe('initialState', () => {
  it('matches snapshot', () => {
    expect(initialState).toMatchSnapshot();
  });
});
