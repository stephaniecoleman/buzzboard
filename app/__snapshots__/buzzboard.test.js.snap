exports[`initialState matches snapshot 1`] = `
Object {
  "currentSound": null,
  "isPlaying": false,
}
`;
